### ISSUE TITLE
Content Delivery API Response for Media URL Returns Absolute URL in RTE Instead of Relative URL

### ISSUE THREAD

**Issue by NijasHameed** [umbraco/Umbraco-CMS#16416]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.2.2

### Bug summary

No configuration is working to remove the absolute URL from media URLs in the Rich Text Editor (RTE) in the content delivery API response.

Tried both below options but not working, still abosolute url appends on media url 
```
"WebRouting": {
		  "UrlProviderMode": "Relative"
	  },
	  "RichTextEditor": {
        "CustomConfig": {
          "relative_urls": true,
		  "convert_urls":false
        }
      },
```

### Steps to reproduce

1. Create a media item within the Rich Text Editor (RTE) on a content node.
2. SET Below Configurations In AppSettings.Json
```
"WebRouting": {
		  "UrlProviderMode": "Relative"
	  },
	  "RichTextEditor": {
        "CustomConfig": {
          "relative_urls": true,
		  "convert_urls":false
        }
      },
```
3. Enable the Content Delivery API.
4. Attempt to fetch the content via Content Delivery API.
5. Notice that the media URL within the RTE returns as an absolute URL instead of a relative URL.

### Expected result / actual result

The Media Url should be Relative in the RTE Response

**Comment by NijasHameed** [comment]
Upon checking through the source code I could see on [ApiRichTextParserBase.cs](https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Infrastructure/DeliveryApi/ApiRichTextParserBase.cs) class the UrlMode set to Absolute by default.
But it should be set to UrlMode.Default right, so then it will map the configuration we have provided on the webrouting configuration for the "UrlProviderMode" on the appsettings.

I tried in my local and it worked as per the configuration provided in the appsettings.json.
